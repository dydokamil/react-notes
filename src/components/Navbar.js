import React from 'react'
import PropTypes from 'prop-types'

export const Navbar = props => (
  <div id="add-remove-navbar" className="my-navbar">
    <button id="new-note-button" onClick={() => props.onNewNote()}>
      <i className="fas fa-plus" />
    </button>
    <button id="remove-note-button" onClick={() => props.onRemoveNote()}>
      <i className="fas fa-trash-alt" />
    </button>
  </div>
)

Navbar.propTypes = {
  onNewNote: PropTypes.func.isRequired,
  onRemoveNote: PropTypes.func.isRequired
}

export default Navbar
