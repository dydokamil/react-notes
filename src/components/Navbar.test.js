import React from 'react'
import { shallow } from 'enzyme'
import { Navbar } from './Navbar'

describe('Navbar component', () => {
  const onNewNoteMock = jest.fn()
  const onRemoveNoteMock = jest.fn()
  const props = { onNewNote: onNewNoteMock, onRemoveNote: onRemoveNoteMock }
  const navbar = shallow(<Navbar {...props} />)
  it('should render the navbar', () => {
    expect(navbar).toMatchSnapshot()
  })
  it('should call `onNewNote` when #new-note-button is pressed', () => {
    navbar.find('#new-note-button').simulate('click')
    expect(onNewNoteMock).toHaveBeenCalled()
  })
  it('should call `onRemoveNote` when #remove-note-button is pressed', () => {
    navbar.find('#remove-note-button').simulate('click')
    expect(onRemoveNoteMock).toHaveBeenCalled()
  })
})
