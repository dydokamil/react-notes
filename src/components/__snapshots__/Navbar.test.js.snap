// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar component should render the navbar 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    onNewNote={[Function]}
    onRemoveNote={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <button
          id="new-note-button"
          onClick={[Function]}
>
          <i
                    className="fas fa-plus"
          />
</button>,
        <button
          id="remove-note-button"
          onClick={[Function]}
>
          <i
                    className="fas fa-trash-alt"
          />
</button>,
      ],
      "className": "my-navbar",
      "id": "add-remove-navbar",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <i
            className="fas fa-plus"
/>,
          "id": "new-note-button",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "fas fa-plus",
          },
          "ref": null,
          "rendered": null,
          "type": "i",
        },
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <i
            className="fas fa-trash-alt"
/>,
          "id": "remove-note-button",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "fas fa-trash-alt",
          },
          "ref": null,
          "rendered": null,
          "type": "i",
        },
        "type": "button",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <button
            id="new-note-button"
            onClick={[Function]}
>
            <i
                        className="fas fa-plus"
            />
</button>,
          <button
            id="remove-note-button"
            onClick={[Function]}
>
            <i
                        className="fas fa-trash-alt"
            />
</button>,
        ],
        "className": "my-navbar",
        "id": "add-remove-navbar",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <i
              className="fas fa-plus"
/>,
            "id": "new-note-button",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "fas fa-plus",
            },
            "ref": null,
            "rendered": null,
            "type": "i",
          },
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <i
              className="fas fa-trash-alt"
/>,
            "id": "remove-note-button",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "fas fa-trash-alt",
            },
            "ref": null,
            "rendered": null,
            "type": "i",
          },
          "type": "button",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
